src/tools/r2rdump/R2RReader.cs

```diff
@@ -113,35 +112,77 @@ public unsafe R2RReader(string filename)
                     R2RMethods = new List<R2RMethod>();
                     for (uint rid = 1; rid <= nMethodEntryPoints; rid++)
                     {
-                        uint offset = 0;
+                        int offset = 0;
                         if (methodEntryPoints.TryGetAt(_image, rid - 1, ref offset))
                         {
-                            R2RMethod method = new R2RMethod(_image, mdReader, methodEntryPoints, offset, rid);
-                            if (method.EntryPointRuntimeFunctionId >= nRuntimeFunctions)
+                            R2RMethod method = new R2RMethod(_image, mdReader, methodEntryPoints, rid, offset);
+
+                            if (method.EntryPointRuntimeFunctionId < 0 || method.EntryPointRuntimeFunctionId >= nRuntimeFunctions)
                             {
-                                throw new BadImageFormatException("Runtime function id out of bounds");
+                                throw new BadImageFormatException("EntryPointRuntimeFunctionId out of bounds");
                             }
                             isEntryPoint[method.EntryPointRuntimeFunctionId] = true;
                             R2RMethods.Add(method);
                         }
                     }
 
+                    // instance method table
+                    R2RSection instMethodEntryPointSection = R2RHeader.Sections[R2RSection.SectionType.READYTORUN_SECTION_INSTANCE_METHOD_ENTRYPOINTS];
+                    int instMethodEntryPointsOffset = GetOffset(instMethodEntryPointSection.RelativeVirtualAddress);
+                    NativeParser parser = new NativeParser(_image, (uint)instMethodEntryPointsOffset);
+                    NativeHashtable instMethodEntryPoints = new NativeHashtable(_image, parser);
+                    NativeHashtable.AllEntriesEnumerator allEntries = instMethodEntryPoints.EnumerateAllEntries();
+                    NativeParser curEntry = allEntries.GetNext();
```

Comment:
Could you rename this variable to make it clear its a parser? "Entry" is somewhat vague